hw3/01-routing.js

```diff
@@ -34,16 +34,62 @@ const server = http.createServer((req, res) => {
     return result;
   };
 
-  if (req.url === '/') {
+  if (req.url === "/") {
     let routeResults = getRoutes();
 
-    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.writeHead(200, { "Content-Type": "text/html" });
     res.write(`<h1>Exercise 01</h1>`);
     res.write(`<ul> ${routeResults} </ul>`);
-    res.end();
+    return res.end();
   }
 
   // Add your code here
+  else if (req.url === `/${routes[0]}`) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write("<h1>Welcome to a new world.</h1>");
+    return res.end();
+  }
+  // Redirect Route
+  else if (req.url === `/${routes[1]}`) {
+    res.writeHead(302, { Location: `/${routes[2]}` });
+    return res.end();
+  }
+  // Redirected Route
+  else if (req.url === `/${routes[2]}`) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write("<h1>You have been redirected.</h1>");
+    return res.end();
+  }
+  // Cache Route
+  else if (req.url === `/${routes[3]}`) {
+    res.setHeader("CacheControl", "max-age=86400");
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    res.write("this resource was cached");
+    return res.end();
+  }
+  // Cookie Route
+  else if (req.url === `/${routes[4]}`) {
+    res.writeHead(200, {
+      "Set-Cookie": "hello=world",
+      "Content-Type": "text/plain",
+    });
+    res.write("cookies... yummm");
+    return res.end();
+  }
+  // Cookie Route
+  else if (req.url === `/${routes[5]}`) {
+    Cookie = req.headers.cookie;
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    if (Cookie === "hello=world") res.write("Yes");
```

Comment:
Hello Brian,

I think it would a good idea to iterate over all the cookies to check if has the 'hello world' cookie, as there can be more than one cookie set in a session.